stups-spring-oauth2-server/src/main/java/org/zalando/stups/oauth2/spring/server/HystrixTokenInfoRequestExecutor.java

```diff
@@ -0,0 +1,50 @@
+package org.zalando.stups.oauth2.spring.server;
+
+import com.netflix.hystrix.HystrixCommand;
+import com.netflix.hystrix.HystrixCommandGroupKey;
+import org.springframework.web.client.RestOperations;
+
+import java.util.Map;
+
+public class HystrixTokenInfoRequestExecutor extends DefaultTokenInfoRequestExecutor {
+
+  public HystrixTokenInfoRequestExecutor(String tokenInfoEndpointUrl) {
+    super(tokenInfoEndpointUrl);
+  }
+
+  public HystrixTokenInfoRequestExecutor(
+      final String tokenInfoEndpointUrl,
+      final RestOperations restOperations) {
+    super(tokenInfoEndpointUrl, restOperations);
+  }
+
+  @Override
+  public final Map<String, Object> getMap(final String accessToken) {
+    return new TokenInfoCommand(this, accessToken).execute();
+  }
+
+  static class TokenInfoCommand extends HystrixCommand<Map<String, Object>> {
+
+    private final HystrixTokenInfoRequestExecutor tokenInfoRequestExecutor;
+
+    private final String accessToken;
+
+    TokenInfoCommand(
+        final HystrixTokenInfoRequestExecutor tokenInfoRequestExecutor,
+        final String accessToken) {
+      super(HystrixCommandGroupKey.Factory.asKey("TokenInfo"));
+      this.tokenInfoRequestExecutor = tokenInfoRequestExecutor;
+      this.accessToken = accessToken;
+    }
+
+    @Override
+    protected Map<String, Object> run() throws Exception {
+      return tokenInfoRequestExecutor.doGetMap(accessToken);
+    }
+
+    @Override
+    protected Map<String, Object> getFallback() {
+      return null;
```

Comment:
I have no idea about hystrixcommands but wouldn't it be better here to return empty map?

Reply:
Based on where it is invoked from I think it should actually return `null`: https://github.com/zalando-stups/stups-spring-oauth2-support/blob/master/stups-spring-oauth2-server/src/main/java/org/zalando/stups/oauth2/spring/server/TokenInfoResourceServerTokenServices.java#L115